clang/lib/Sema/SemaType.cpp

```diff
@@ -7963,6 +7967,148 @@ static Attr *getCCTypeAttr(ASTContext &Ctx, ParsedAttr &Attr) {
   llvm_unreachable("unexpected attribute kind!");
 }
 
+ExprResult Sema::ActOnEffectExpression(Expr *CondExpr, FunctionEffectMode &Mode,
+                                       bool RequireConstexpr) {
+  // see checkFunctionConditionAttr, Sema::CheckCXXBooleanCondition
+  if (RequireConstexpr || !CondExpr->isTypeDependent()) {
+    ExprResult E = PerformContextuallyConvertToBool(CondExpr);
+    if (E.isInvalid())
```

Comment:
```suggestion
    if (!E.isUsable())
```
`!isInvalid()` doesn’t mean you have an expression, unfortunately; the expression in the result can be null; `isUsable()` checks for that too.

Reply:
This was copied from `Sema::CheckCXXBooleanCondition`; is it incorrect too?

Reply:
It might be fine in that case; the main reason why I suggested `isUsable()` is mostly ‘just in case’, i.e. if I don’t know off the top of my head whether there are any code paths there that could possibly lead to something being unset, then I generally prefer using `isUsable()` just in case.